### PR TITLE
Add a prop to SelectionMesh to run a callback on selection

### PR DESCRIPTION
**Pull request by loichuder**

### CODE REVIEW

File: apps/storybook/src/VisCanvasInteraction.stories.tsx [hunk 1]
```diff
@@ -18,31 +21,56 @@ interface TemplateProps {
...
+            onSelection={(start, end) => setSelectedVectors([start, end])}
+          />
+        )}
+        {selection === 'rectangle' && (
+          <RectSelectionMesh
+            onSelection={(start, end) => setSelectedVectors([start, end])}
+          />
+        )}
+      </VisCanvas>
+    </>
```
Comment (loichuder) [inline apps/storybook/src/VisCanvasInteraction.stories.tsx:73]: The template of the story is becoming very "selection-specific". Do you think I should move it to a new file `VisCanvasSelection.stories` ?

File: packages/lib/src/vis/shared/SelectionMesh.tsx [hunk 2]
```diff
@@ -57,8 +58,11 @@ function SelectionMesh(props: Props) {
       (target as Element).releasePointerCapture(pointerId);
       setEndPoint(point);
       setDrag(false);
+      if (onSelection && startPoint && endPoint) {
+        onSelection(startPoint, endPoint);
+      }
```
Comment (loichuder) [inline packages/lib/src/vis/shared/SelectionMesh.tsx:63]: I thought it was best to run the callback after `onPointerUp` rather than use a `useEffect`.

Comment (axelboc) [reply packages/lib/src/vis/shared/SelectionMesh.tsx:63]: Yep, that's great 👍

File: apps/storybook/src/VisCanvasInteraction.stories.tsx (hunk 1 above)
Comment (axelboc) [reply apps/storybook/src/VisCanvasInteraction.stories.tsx:73]: Yeah, with the local state, it's looking a bit bloated, you're right.

**Review by axelboc — APPROVED** [review]

**Pull request merged**